internals/controllers/cappPlacement_controller.go

```diff
@@ -44,19 +48,22 @@ func (r *ServicePlacementReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	}
 	placementRef := capp.Spec.Site
 	if placementRef == "" || slices.Contains(r.Placements, placementRef) {
-		cluster, err := r.pickDecision(capp, l, ctx)
+		cluster, err := r.pickDecision(capp, logger, ctx)
 		if err != nil {
+			logger.Error(err, fmt.Sprintf("Failed to pick managed cluster for placement %s.", placementRef))
 			return ctrl.Result{}, err
 		}
 		if cluster == "requeue" {
+			logger.Info(fmt.Sprintf("requeuing capp %s, waiting for PlacementDecision to be satisfied.", capp.Name))
+			r.EventRecorder.Event(&capp, "Warning", "PlacementDecisionNotSatisfied", fmt.Sprintf("Failed to schedule capp %s to managed cluster. PlacementDecision with optional clusters was not found for placement %s.", capp.Name, placementRef))
 			return ctrl.Result{RequeueAfter: 10 * time.Second * 2}, nil
 		}
 		placementRef = cluster
 	}
 	if err := utils.UpdateCappDestination(capp, placementRef, ctx, r.Client); err != nil {
-		l.Error(err, "unable to update capp")
-		return ctrl.Result{}, err
+		return ctrl.Result{}, fmt.Errorf("Unable to update capp with selected cluster %s", err.Error())
```

Comment:
There's a lack of consistency: at times it's written as `capp` and at times as `Capp`. I think we should go with `Capp` all across the logs and events